src/parsers/toml.rs

```diff
@@ -14,3 +14,36 @@ impl Parser for Toml {
         Ok(toml::from_str(s)?)
     }
 }
+
+/// Toml with unknown field handler support
+///
+/// This parser will ignore unknown fields and call the handler with the path of the unknown field.
+pub struct TomlIgnored {
```

Comment:
Hi, I have a feeling that this API doesn't belong to serfig. Maybe we can move it to downstream?

Reply:
make sense, let me put it in the downstream first

Reply:
Thank you for this. I believe other parts of this PR is good to go.

Reply:
let me rename this pr's title to reduce the scope of change